### PR TITLE
RET-4142 : Add a null check to getHearingType

### PR DESCRIPTION
**Pull request by ss-evoco**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by brendanmchugh — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/et/helper/mapping/CaseFlagsMapping.java [hunk 1]
```diff
@@ -1,57 +1,54 @@
...
-    private static final String ACTIVE_STATUS = "Active";
-
     private CaseFlagsMapping() {
-        //NO-OP
+
     }
 
     public static CaseFlags getCaseFlags(CaseData caseData) {
-        List<CaseFlagsType> allActiveFlags = getAllActiveFlags(caseData);
+        List<PartyFlags> flags = new ArrayList<>();
```
Comment (marcjudge) [inline src/main/java/uk/gov/hmcts/reform/et/helper/mapping/CaseFlagsMapping.java:20]: Add JavaDocs

File: src/main/java/uk/gov/hmcts/reform/et/helper/mapping/HearingsCaseMapping.java [hunk 2]
```diff
@@ -4,70 +4,88 @@
...
-        CaseCategory caseCategory = new CaseCategory();
-        caseCategory.setCategoryType(DEFAULT_CATEGORY);
-        caseCategory.setCategoryValue(DEFAULT_CATEGORY);
-        caseCategory.setCategoryParent(DEFAULT_CATEGORY);
-
-        List<CaseCategory> resultList = new ArrayList<>();
-        resultList.add(caseCategory);
-
-        return resultList;
+        String caseType = "BHA1-EMT";
```
Comment (marcjudge) [inline src/main/java/uk/gov/hmcts/reform/et/helper/mapping/HearingsCaseMapping.java:56]: Add JavaDocs

File: src/main/java/uk/gov/hmcts/reform/et/helper/mapping/HearingsPartyMapping.java [hunk 3]
```diff
@@ -3,119 +3,125 @@
...
-        //NO-OP
-    }
 
-    public static List<PartyDetails> buildPartyObjectForHearingPayload(
-        CaseData caseData,
-        List<RespondentSumTypeItem> respondents,
-        List<RepresentedTypeRItem> representatives) {
+    }
 
+    public static List<PartyDetails> buildPartyObjectForHearingPayload(CaseData caseData) {
```
Comment (marcjudge) [inline src/main/java/uk/gov/hmcts/reform/et/helper/mapping/HearingsPartyMapping.java:32]: Add JavaDocs

File: src/main/java/uk/gov/hmcts/reform/et/service/ServiceHearingsService.java [hunk 4]
```diff
@@ -24,22 +25,49 @@ public class ServiceHearingsService {
...
 
-    private final ReferenceDataServiceHolder referenceDataServiceHolder;
+    @Value("${et.hmctsServiceId}")
+    private String hmctsServiceId;
 
-    public ServiceHearingValues getServiceHearingValues(
-        String authorization,
-        ServiceHearingRequest request
-    ) throws GetCaseException {
+    public ServiceHearingValues getServiceHearingValues(String authorization, ServiceHearingRequest request)
```
Comment (marcjudge) [inline src/main/java/uk/gov/hmcts/reform/et/service/ServiceHearingsService.java:31]: Add JavaDocs

File: src/main/java/uk/gov/hmcts/reform/et/service/ServiceLinkedCasesService.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+import uk.gov.hmcts.reform.et.helper.mapping.CaseDataMapping;
+import uk.gov.hmcts.reform.et.model.service.ServiceHearingRequest;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class ServiceLinkedCasesService {
+    private final CaseService caseService;
+
+    public ListTypeItem<CaseLink> getServiceLinkedCases(String authorization, ServiceHearingRequest request)
```
Comment (marcjudge) [inline src/main/java/uk/gov/hmcts/reform/et/service/ServiceLinkedCasesService.java:20]: Add JavaDocs

**Review by marcjudge — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/et/helper/mapping/HearingsDetailsMapping.java [hunk 6]
```diff
@@ -1,57 +1,41 @@
...
-
-    public static String getHearingType(String hearingRequest, List<HearingTypeItem> hearingCollection) {
-        String hearingType = null;
-        for (HearingTypeItem hearingItem : hearingCollection) {
-            if (hearingItem.getId().equals(hearingRequest)) {
-                hearingType = hearingItem.getValue().getHearingType();
-            }
-        }
-        return hearingType;
+    public static Integer getHearingDuration() {
```
[the remark this answers is not in the document]
Comment (JackReevies) [reply src/main/java/uk/gov/hmcts/reform/et/helper/mapping/HearingsDetailsMapping.java:37]: No, I don't think there is. Good spot, I'll simplify

**Review by marcjudge — DISMISSED** [review]

**Review by marcjudge — APPROVED** [review]

**Pull request merged**